core/src/main/scala/org/bitcoins/core/wallet/utxo/ConditionalPath.scala

```diff
@@ -0,0 +1,44 @@
+package org.bitcoins.core.wallet.utxo
+
+/** Represents the spending branch being taken in a ScriptPubKey's execution
+  *
+  * If you over-specify a path, such as giving a condition where none is needed,
+  * then the remaining over-specified path will be ignored.
+  *
+  * For example, if you wanted to spend a ConditionalScriptPubKey(P2PK1, P2PK2)
+  * (which looks like OP_IF <P2PK1> OP_ELSE <P2PK2> OP_ENDIF) with the P2PK1 case,
+  * then you would construct a ConditionalSpendingInfo using nonNestedTrue as your
+  * ConditionalPath. Otherwise if you wanted to use P2PK2 you would use nonNestedFalse.
+  */
+sealed trait ConditionalPath {
```

Comment:
I don't love the names for the subtypes of `ConditionalPath` so feel free to change, but 

```
sealed trait ConditionalPath

sealed abstract class Leaf extends ConditionalPath

sealed abstract class Node extends ConditionalPath {
  def head: Boolean
}

object ConditionalPath {
  case object NoConditionsLeft extends Leaf

  case class ConditionTrue(nextCondition: ConditionalPath)
      extends Node {
    override val head: Boolean = true
  }
  case class ConditionFalse(nextCondition: ConditionalPath)
      extends Node {
    override val head: Boolean = false
  }
```

is what this should probably look like